sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -74,13 +74,71 @@ object RDDConversions {
   }
 }
 
+private[sql] object ExternalRDD {
+
+  def apply[T: Encoder](rdd: RDD[T])(session: SparkSession): LogicalPlan = {
```

Comment:
There is probably no reason to use multiple parameter lists here.  We sometimes use it for case classes so that arguments that should not effect equality are not include in the generated `equals` method.